core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -68,13 +82,26 @@
     private static final Logger LOG = LoggerFactory.getLogger(Factory.class);
     private static final AtomicBoolean CONF_LOGGED = new AtomicBoolean(false);
 
+    protected static final FileSystem.Cache FILESYSTEM_CACHE = new FileSystem.Cache();
+
     private Factory() {} // prevent instantiation
 
-    public static FileSystem get(AlluxioConfiguration alluxioConf) {
-      return get(FileSystemContext.create(alluxioConf));
+    public static FileSystem get() {
+      return get(new Subject()); // Use empty subject
+    }
+
+    public static FileSystem get(Subject subject) {
+      Preconditions.checkNotNull(subject, "subject");
+      AlluxioConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
```

Comment:
Consider referencing a single copy of the default configuration? This can reduce the number of times validation/default fetching, and object creation needs to occur.

Reply:
Well the main point of removing the singleton configuration was to not be able to easily reference a static configuration anywhere. Introducing some single static configuration object, I believe, goes against the point of removing the configuration in the first place. Also users really shouldn't be calling `get` constantly. We should encourage sharing the reference rather than calling get constantly.

Either way, I see the usefulness in adding something to `ConfigurationUtils` that returns a static reference to an object which avoids validation but I'm reluctant. If you really want it I will add it.

Reply:
If creating the new instance is cheap, its not a problem. 

For singleton-like objects, I think if they are immutable and we are providing them under the hood (not like accessing a static member) it is acceptable.